src/realm/object-store/dictionary.cpp

```diff
@@ -67,6 +67,44 @@ Results Dictionary::get_values() const
     return as_results();
 }
 
+class NotificationHandler {
+public:
+    NotificationHandler(realm::Dictionary& dict, Dictionary::CBFunc cb)
+        : m_dict(dict)
+        , m_cb(cb)
+    {
+    }
+    void before(CollectionChangeSet const&) {}
+    void after(CollectionChangeSet const& c)
+    {
+        DictionaryChangeSet changes;
+        for (auto ndx : c.deletions.as_indexes()) {
+            changes.deletions.push_back(ndx);
+        }
+        for (auto ndx : c.insertions.as_indexes()) {
+            changes.insertions.push_back(m_dict.get_key(ndx));
+        }
+        for (auto ndx : c.modifications.as_indexes()) {
```

Comment:
This probably needs to be modifications_new.

Reply:
Ok. Changed!